TASK-30 Completed: Room Chat UI and integration with signalR gameHub service

Asafyaco opened the pull request:
Created chat Real-time **responsible** ( for all devices ) for all the players on game.page:

* The chat is availabile **only** for the players who are in the same room - **not for everyone** (filer by roomId)
* Button icon to open the chat 
* Chat wrapper that icludes:
* * Input - for the message
* * Button - for send the message ( There's an option to click on send button or **click enter** to send the message ) 
* The messegae includes- first & last name of the player and the message he sent
* Chat scroll automatically to the bottom if there's a new message 
* Connection to signalR with hubConnection and created connection "newChatMessage" on useEffect

**Examples:**
![image](https://user-images.githubusercontent.com/93646445/165150518-e6d68331-cd01-4236-92b0-59cb36404433.png)
![image](https://user-images.githubusercontent.com/93646445/165150597-7ec88b54-a2a4-4930-a80e-941b3ea61f30.png)
![image](https://user-images.githubusercontent.com/93646445/165150757-14b0eee5-7991-4e29-96c1-499ddaefd0bb.png)

Display for an iPhone:
![image](https://user-images.githubusercontent.com/93646445/165150976-fe1139d3-4138-48ed-883f-7624a21ac7da.png)

FadiFarah committed:
TASK-30 Completed: Room Chat UI and integration with signalR gameHub service

FadiFarah commented on src/pages/game/components/chat/chat-messages.component.jsx:
```diff
@@ -0,0 +1,82 @@
+  };
+
+  const open = props.isOpen;
+  const id = open ? "simple-popover" : undefined;
```
remove all the code from line 17 to here(28). Not needed. We are already sending all the needed data from the parent.

FadiFarah commented on src/pages/game/game.page.jsx:
```diff
@@ -147,6 +206,32 @@ const GamePage = () => {
+            <ion-icon
+              name="chatbubbles-outline"
+              // onClick={(e) => setIsChatDisplay(!isChatDisplay)}
+              onClick={(e) => handleClick(e)}
```
remove the commented onClick. Also move the current onClick to the ion-fab-button instead of the ion-icon. Currently, only if we press on the icon inside the fab button, the chat box opens, and not the whole button itself, which is a bad user experience.

Asafyaco replied:
Oh ok, I got it. Changed.

FadiFarah reviewed (changes requested):
Some changes needed.

Asafyaco commented:
Changes made by comments.
- Deleted code unnecessary
- Change onClick chat locate- on icon-fab-button
- Added placeholder to the TextField of the chat message - 
![image](https://user-images.githubusercontent.com/93646445/165503928-99453deb-c356-4660-8acd-0695572da109.png)

FadiFarah reviewed (approved):
great!
